### PR TITLE
Add extra typehinting tests

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds some extra tests around enumeration type hinting in collections, also covers the built-in `frozenset`case that didn't have a test. 

Part of #29135

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
R: @damccorm

File: sdks/python/apache_beam/typehints/native_type_compatibility_test.py [hunk 1]
```diff
@@ -122,20 +129,22 @@ def test_convert_to_beam_type(self):
...
-                    (
-                        'builtin nested tuple',
-                        tuple[str, list],
-                        typehints.Tuple[str, typehints.List[typehints.Any]],
-                    )]
+      test_cases = [
+          ('builtin dict', dict[str, int], typehints.Dict[str, int]),
+          ('builtin list', list[str], typehints.List[str]),
+          ('builtin tuple', tuple[str],
+           typehints.Tuple[str]), ('builtin set', set[str], typehints.Set[str]),
```
Comment (damccorm) [inline sdks/python/apache_beam/typehints/native_type_compatibility_test.py:136]: Nit: weird spacing here
```suggestion
          ('builtin tuple', tuple[str], typehints.Tuple[str]),
          ('builtin set', set[str], typehints.Set[str]),
```

**Review by damccorm — APPROVED** [review]
LGTM other than one nit

File: sdks/python/apache_beam/typehints/native_type_compatibility_test.py (hunk 1 above)
Comment (jrmccluskey) [reply sdks/python/apache_beam/typehints/native_type_compatibility_test.py]: Hmm yapf did something weird here for sure, changed

Comment (damccorm) [reply sdks/python/apache_beam/typehints/native_type_compatibility_test.py]: Lol, the formatter really doesn't like it I guess. Well, we tried - feel free to revert/merge or to find a way to fix forward the formatting

**Comment by jrmccluskey** [comment]
Run Python_PVR_Flink PreCommit

**Comment by jrmccluskey** [comment]
Run Python_PVR_Flink PreCommit

**Comment by jrmccluskey** [comment]
Run Python_PVR_Flink PreCommit

**Comment by jrmccluskey** [comment]
Run Python_Transforms PreCommit

**Comment by jrmccluskey** [comment]
Run Python_PVR_Flink PreCommit

**Comment by jrmccluskey** [comment]
Run Python_PVR_Flink PreCommit

**Pull request merged**